cv_next/server/api/cvs.ts

```diff
@@ -1,277 +1,179 @@
- *
- * @param {Categories.category} category - The category to filter CVs by.
- * @param {boolean} filterOutDeleted - Whether to exclude deleted CVs from the result. Defaults to true.
- * @return {Promise<CvModel[] | null>} - A promise that resolves to an array of CV models or null.
- */
 export async function getAllCvsByCategory(
   category: Categories.category,
   filterOutDeleted: boolean = true
 ): Promise<CvModel[] | null> {
-  return _getAllCvsByQueryFilter(
-    where("categoryID", "==", category),
-    filterOutDeleted
-  )
+  try {
+    const supabase = SupabaseHelper.getSupabaseInstance()
+    let query = supabase.from("cvs").select("*").eq("category_id", category)
+
+    if (filterOutDeleted) {
+      query = query.eq("deleted", false)
+    }
+
+    const { data: cvs, error } = await query
+
+    if (error) {
+      MyLogger.logInfo("Error @ getAllCvsByCategory", error)
+      return null
+    }
+
+    return cvs as CvModel[]
+  } catch (error) {
+    MyLogger.logInfo("Error @ getAllCvsByCategory", error)
+    return null
+  }
 }
 
 /**
- * Retrieves paginated CVs from the database.
+ * Retrieves a paginated list of CVs based on the provided page number.
  *
- * @param {boolean} filterOutDeleted - Whether to filter out deleted CVs.
- * @param {string?} lastId - optionally pass the last cv id to start the query after. 
- * @return {Promise<CvModel[] | null>} - A promise that resolves to an array of CV models or null.
+ * @param {boolean} filterOutDeleted - Indicates whether deleted CVs should be filtered out.
+ * @param {number} page - The page number for pagination.
+ * @return {Promise<CvModel[] | null>} A Promise that resolves with an array of CvModel or null.
  */
 export async function getPaginatedCvs(
   filterOutDeleted: boolean = true,
-  lastId?: string
-): Promise<CvModel[] | null> {
-  let res = getPaginatedCvsByQueryFilter(undefined, filterOutDeleted, lastId)
-  MyLogger.logDebug(
-    `requested all cvs(pagination) - last cv id: ${lastId}`
-  )
-  return res
+  page: number = Definitions.PAGINATION_INIT_PAGE_NUMBER,
+  filters?: filterObj,
+): Promise<PaginatedCvsModel | null> {
+  try {
+    const from = page * Definitions.CVS_PER_PAGE
+    const to = page ? from + Definitions.CVS_PER_PAGE : Definitions.CVS_PER_PAGE
+
+    const supabase = SupabaseHelper.getSupabaseInstance()
+    let query = supabase
+      .from("cvs")
+      .select("*")
+      .range(from, to - 1)
+
+    console.log("filters", filters)
+    if(filters) {
+      if(filters.searchValue){  
+      query = query.textSearch('description', filters.searchValue, {
+        type:"plain"
+      })
+      }
+      if(filters.categoryId){  
+        console.log("catagory id:", filters.categoryId)
+        query = query.eq("category_id", filters.categoryId)
+      }
+    }    
+
+    if (filterOutDeleted) {
+      query = query.eq("deleted", false)
+    }
+
+    const { data: cvs, error } = await query
+    console.log("cvs:", cvs?.map(cv => cv.category_id))
```

Comment:
Don't forget to delete console logs before pushing.
Also @Aradon267 I'm pretty sure there's either a commit hook or a github actions rule to highlight these logs.

Reply:
Would it be better to change it to the logger that is present in the project? Or just remove it completely? I'll check about the hook

Reply:
I'll add an eslint commit rule